src/itwinai/torch/loggers.py

```diff
@@ -0,0 +1,219 @@
+                Specifies which checkpoints to log.
+                If "all", logs all checkpoints; if True, logs the best k checkpoints according
+                to the specifications given as `save_top_k` in the Lightning ModelCheckpoint;
+                if False, does not log checkpoints.
+        """
+        self.itwinai_logger = itwinai_logger
+
+        self._log_model = log_model
+        self._logged_model_time = {}
+        self._checkpoint_callback = None
+
+        self._initialized = False
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return the experiment name."""
+        self.experiment.experiment_id
+
+    @property
+    def version(self) -> Optional[Union[int, str]]:
+        """Return the experiment version."""
+        self.experiment.run_id
+
+    @property
+    @override
+    @rank_zero_only
+    def save_dir(self) -> Optional[str]:
+        """Return the directory where the logs are stored."""
+        return self.experiment.savedir
+
+    @property
+    @rank_zero_experiment
+    def experiment(self) -> ItwinaiBaseLogger:
+        """Lightning Logger function.
+        Initializes and returns the itwinai Logger context for experiment tracking.
+
+        Returns:
+            Logger: The itwinai logger instance.
+        """
+        if not self._initialized:
+            # With the rank_zero_experiment decorators the rank will always be 0
+            self.itwinai_logger.create_logger_context(rank=0)
+            self._initialized = True
+
+        return self.itwinai_logger
+
+    @override
+    @rank_zero_only
+    def finalize(self, status: str) -> None:
+        """Lightning Logger function.
+        Logs any remaining checkpoints and closes the logger context.
+
+        Args:
+            status (str): Describes the status of the training (e.g., 'completed', 'failed').
+            The status is not needed for this function but part of the parent classes'
+                (LightningLogger)
+            finalize functions signature, and therefore must be propagated here.
+        """
+        if not self._initialized:
+            return
+
+        # Log checkpoints if the last checkpoint was saved but not logged
+        if self._checkpoint_callback:
+            self._scan_and_log_checkpoints(self._checkpoint_callback)
+
+        self.experiment.destroy_logger_context()
+
+    @override
+    @rank_zero_only
+    def log_metrics(
+        self, metrics: Dict[str, float], step: Optional[int] = None
+    ) -> None:
+        """Lightning Logger function.
+        Logs the given metrics and is usually called by the Lightning Trainer.
+
+        Args:
+            metrics (Dict[str, float]): Dictionary of metrics to log.
+            step (Optional[int], optional): Training step associated with the metrics.
+                Defaults to None.
+        """
+        for key, value in metrics.items():
+            self.experiment.log(item=value, identifier=key, kind="metric", step=step)
```

Comment:
This is a very small thing and I am not even sure about it, but it could be nicer name the variables differently than "key" and "value" here, since you are using them for something else. E.g. you could say
```
for identifier, item in metrics.items(): 
```
That being said, I am not super opposed to the idea of calling them "key" and "value" since it is a dictionary, but I think descriptive names are always better. Really not that important in this two-line example though, haha, so you can leave it if you want.